src/main/java/net/fabricmc/nameproposal/field/nameprovider/FieldNameProvider.java

```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright (c) 2023 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.nameproposal.field.nameprovider;
+
+import com.mojang.serialization.Codec;
+
+import net.fabricmc.nameproposal.field.FieldData;
+
+public abstract class FieldNameProvider {
```

Comment:
Would this be better as an interface, then allowing the children to be a record? Or am I missing something?

Reply:
Fixed in d25a158f2a2110c33990af43fbb1a4f3cfc7f903